webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/smartling/SmartlingResultProcessor.java

```diff
@@ -0,0 +1,105 @@
+package com.box.l10n.mojito.service.thirdparty.smartling;
+
+import com.amazonaws.SdkClientException;
+import com.box.l10n.mojito.service.blobstorage.Retention;
+import com.box.l10n.mojito.service.blobstorage.s3.S3BlobStorage;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+@ConditionalOnProperty(value = "l10n.ThirdPartyTMS.impl", havingValue = "ThirdPartyTMSSmartling")
+@Component
+public class SmartlingResultProcessor {
+
+    static Logger logger = LoggerFactory.getLogger(SmartlingResultProcessor.class);
+
+    private static final String NAME_FORMAT = "%s_%s_%s.zip";
+
+    @Autowired
+    S3BlobStorage s3BlobStorage;
+
+    public String processPush(List<SmartlingFile> files,
+                              SmartlingOptions options) {
+        return processAction(files, options, "push");
+    }
+
+    public String processPushTranslations(List<SmartlingFile> files,
+                                          SmartlingOptions options){
+        return processAction(files, options, "push_translations");
+    }
+
+    private String processAction(List<SmartlingFile> files,
+                                 SmartlingOptions options,
+                                 String action) {
+        String result = null;
+
+        if (!files.isEmpty() && options.isDryRun() && s3BlobStorage != null){
+            try {
+                result = uploadFiles(files, options.getRequestId(), action);
+                logger.info("{} result for request id {} uploaded to: {}", action, options.getRequestId(), result);
+            } catch (IOException | SdkClientException e) {
+                logger.error(String.format("An error ocurred when uploading a %s result zip file", action), e);
```

Comment:
should the exception be re-thrown? what happens if it fails?

Reply:
That's why I just logged the exception. Considering this a side-effect, if there's a failure when uploading to S3 doesn't mean we should error out the entire sync process.

Reply:
well you're doing a dry run, so it shouldn't actually sync and just produce files for testing. Looks like that won't be possible so it should fail hard. 

This won't be done for normal sync will it?

Reply:
That's a good point. Updated.

And no, we won't be doing for normal sync, just when doing a dry run.

Reply:
can you also pass the error message in the runtimeexception along with the cause?

Reply:
Done